File: backend/app/models/schemas/retrieval.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel


class QueryRequest(BaseModel):
    query: str
    num_results: int = 3
    url_prefix: str = None


class RetrievalResult(BaseModel):
    answer: str
    sources: List[Dict[str, Any]]
    query_time_seconds: float
    follow_up_questions: List[str]


class RetrievalRequest(BaseModel):
    query: str
    num_results: int = 3
    url_prefix: Optional[str] = None

File: frontend/src/services/retrieval.ts
import { fetchApi } from '../utils/api-client';
import { API_CONFIG } from '../config/api';

export interface RetrievalResult {
  answer: string;
  sources: Array<{
    url: string;
    text: string;
  }>;
  follow_up_questions: string[];
}

interface RetrievalRequest {
  query: string;
  num_results: number;
  url_prefix?: string;
}

interface RetrievalResponse {
  status: string;
  data: RetrievalResult;
}

export const retrievalService = {
  search: (data: RetrievalRequest) =>
    fetchApi<RetrievalResponse>(API_CONFIG.ENDPOINTS.RETRIEVAL, {
      method: 'POST',
      body: JSON.stringify(data),
    }),
}; 